Список школ, упорядоченный по среднему баллу участников

Maksonchik88 opened the pull request:
https://informatics.msk.ru/mod/statements/view.php?id=5763&chapterid=111352#1

бесит сука! Оставлю пока так. Пока не разъебал комп!

Maksonchik88 committed:
абсолютно не пойму пока, как решитть, поэтому отложу её. Просто бесит, сукаgit add list_of_schools_sorted_by_GPA.py

iliadmitriev commented on files/list_of_schools_sorted_by_GPA.py:
```diff
@@ -0,0 +1,43 @@
+    middle_score = sum(score)/len(score)
+    data[school] = middle_score
+# sort scores in descending order
+scores_sorted = sorted(data.values(), reverse=True)
+schools_sorted = sorted(data.keys())
```
А зачем ты сделал отдельно отсортированные списки школ и баллов. Смысл в том чтобы у тебя структура была не разрывной.

iliadmitriev commented on files/list_of_schools_sorted_by_GPA.py:
```diff
@@ -0,0 +1,43 @@
+scores_sorted = sorted(data.values(), reverse=True)
+schools_sorted = sorted(data.keys())
+
+data_reverse = {score: school for school, score in data.items()}
```
Так конечно можно сделать, но могут быть коллизии. Допустим если у тебя будут две школы с одинаковыми баллами.
Тут у тебя в качестве ключа должен быть балл, по которому ты потом будешь сортировать. А в качестве значения - список школ (а не одна школа).
К стати список школ тоже надо сортировать (по условию задачи)

iliadmitriev reviewed (changes requested):
Неплохо начал. Попробуй развить мысль.

Maksonchik88 committed:
Создали список уникальных отсортированных средних баллов по убыванию
Создали словарь, где ключами являются средние баллыБ а значениями - списки школ
Вывели результат, перебирая отсортированный список средних баллов и сортируя список школ
